info.limpet.stackedcharts.ui/src/info/limpet/stackedcharts/ui/editor/figures/ChartFigure.java

```diff
@@ -14,28 +18,56 @@
 
 public class ChartFigure extends RectangleFigure
 {
-  private Label chartNameLabel;
-  private JFreeChartFigure chartFigure;
+  private final Label chartNameLabel;
+  private final JFreeChartFigure chartFigure;
   private static volatile Font boldFont;
 
-  public ChartFigure(Chart chart)
+  public ChartFigure(final Chart chart, final ActionListener deleteListener)
   {
+    setPreferredSize(-1, 200);
     setBackgroundColor(Display.getDefault().getSystemColor(SWT.COLOR_WHITE));
     setOutline(false);
     setLayoutManager(new BorderLayout());
+    final RectangleFigure rectangleFigure = new RectangleFigure();
+
+    rectangleFigure.setOutline(false);
+    final FlowLayout layout = new FlowLayout(true);
+    layout.setMinorAlignment(OrderedLayout.ALIGN_CENTER);
+    layout.setMajorAlignment(OrderedLayout.ALIGN_CENTER);
+    rectangleFigure.setLayoutManager(layout);
+
     chartNameLabel = new Label();
-    add(chartNameLabel, BorderLayout.TOP);
+    rectangleFigure.add(chartNameLabel);
+    final Button button = new Button("X");
+    button.setToolTip(new Label("Remove this chart from the chart set"));
+    button.addActionListener(deleteListener);
+    rectangleFigure.add(button);
+
+    add(rectangleFigure, BorderLayout.TOP);
+
     chartFigure = new JFreeChartFigure(chart);
     add(chartFigure, BorderLayout.CENTER);
   }
 
-  public void setName(String name)
+  @Override
+  protected void paintClientArea(final Graphics graphics)
+  {
+    super.paintClientArea(graphics);
+    graphics.setForegroundColor(Display.getDefault().getSystemColor(
+        SWT.COLOR_DARK_GRAY));
+
+    final Rectangle clientArea = getClientArea();
+    graphics.drawLine(clientArea.getBottomLeft().getTranslated(0, -1),
+        clientArea.getBottomRight().getTranslated(0, -1));
+  }
+
+  public void setName(final String name)
   {
     chartNameLabel.setText("Chart: " + name);
     // cache font for AxisNameFigure
     if (boldFont == null)
     {
-      FontData fontData = chartNameLabel.getFont().getFontData()[0];
+      final FontData fontData = chartNameLabel.getFont().getFontData()[0];
       boldFont =
```

Comment:
right now we statically keep boldFont. will port it to activator base Font Registry